lib/common/hostinfo.c

```diff
@@ -99,9 +99,9 @@ static void create_lookup_thread(void)
     pthread_attr_setstacksize(&attr, 100 * 1024);
     if ((ret = pthread_create(&tid, NULL, lookup_thread_main, NULL)) != 0) {
         if (queue.num_threads == 0) {
-            h2o_error_printf("failed to start first thread for getaddrinfo:%s\n", strerror(ret));
-            abort();
+            h2o_fatal("failed to start first thread for getaddrinfo: %d", ret);
```

Comment:
I think we'd prefer reporting the value returned by `strerror` in some ways. Otherwise, users of H2O cannot figure out what's going on themselves (and we'd be required to spend time responding to the issues they open).

The same applies to other changes.

Reply:
Agree，after #2022 merged， i will fix this by using `h20_strerror_r`.

Reply:
@kazuho Fixed. please refer to dec8ec5